watcher/watcher.go

```diff
@@ -375,54 +393,81 @@ func (w *Watcher) AutoScale() {
 	}
 }
 
+func (w *Watcher) passedThreshold(scaleType string, autoScalingInfo *models.AutoScaling) bool {
+	if scaleType == "up" {
+		totalPoints := autoScalingInfo.Up.Trigger.Time/w.AutoScalingPeriod + 1
+		pointsAbove := w.ScaleUpInfo.PointsAboveThreshold
+		return pointsAbove*100 > totalPoints*autoScalingInfo.Up.Trigger.Percentage
+	} else if scaleType == "down" {
+		totalPoints := autoScalingInfo.Down.Trigger.Time / w.AutoScalingPeriod
+		pointsAbove := w.ScaleDownInfo.PointsAboveThreshold
+		return pointsAbove*100 > totalPoints*autoScalingInfo.Down.Trigger.Percentage
+	}
+	return false
+}
+
 func (w *Watcher) checkState(
 	autoScalingInfo *models.AutoScaling,
 	roomCount *models.RoomsStatusCount,
 	scheduler *models.Scheduler,
 	nowTimestamp int64,
-) (bool, bool, bool) {
-	inCooldownPeriod := false
+) (bool, bool, bool) { //shouldScaleUp, shouldScaleDown, changedState
+	var shouldScaleUp, shouldScaleDown, changedState bool
+	inSync := true
+	now := time.Unix(nowTimestamp, 0)
 
 	if scheduler.State == models.StateCreating || scheduler.State == models.StateTerminating {
-		return false, false, false
+		return false, false, changedState
 	}
 
 	if roomCount.Total() < autoScalingInfo.Min {
-		return true, false, false
+		return true, false, changedState
 	}
 
 	if 100*roomCount.Ready < (100-autoScalingInfo.Up.Trigger.Usage)*roomCount.Total() {
+		inSync = false
```

Comment:
is this right? shouldn't we set inSync to false only when w.passedThreshold("up", autoScalingInfo)==true ?